diff --git a/r8127.h b/r8127.h
index cf9eb4f..0205179 100755
--- a/r8127.h
+++ b/r8127.h
@@ -648,6 +648,8 @@ r8127  Copyright (C) 2026 Realtek NIC software team <nicfae@realtek.com> \n \
 This program comes with ABSOLUTELY NO WARRANTY; for details, please see <http://www.gnu.org/licenses/>. \n \
 This is free software, and you are welcome to redistribute it under certain conditions; see <http://www.gnu.org/licenses/>. \n"
 
+#define RTL8127_DEBUG
+
 #ifdef RTL8127_DEBUG
 #define assert(expr) \
         if(!(expr)) {                   \
diff --git a/r8127_n.c b/r8127_n.c
index 2279527..2d26984 100755
--- a/r8127_n.c
+++ b/r8127_n.c
@@ -16843,6 +16843,9 @@ static void rtl8127_cancel_schedule_dash_work(struct rtl8127_private *tp)
 
 static void rtl8127_schedule_sds_sfp_cap_work(struct rtl8127_private *tp)
 {
+        if (tp->HwFiberModeVer != FIBER_MODE_RTL8127ATF)
+                return;
+
         set_bit(R8127_FLAG_TASK_SDS_SFP_CAP_PENDING, tp->task_flags);
         schedule_delayed_work(&tp->sds_sfp_cap_task, RTL8127_CHECK_SDS_SFP_CAP_TIMEOUT);
 }
